docs/internal-states.md

```diff
@@ -0,0 +1,178 @@
+# Windmill Internal State 
+
+**Internal States** allow you to save state to the Windmill cloud for use in your scripts and flows. Internal states allow you to set, update and retrieve states.  
```

Comment:
remove "Windmill Cloud", it also works for self-hosted.

Also add a bit of context:
"It is often useful to have a state that persists across an execution of the same script of the flow. For instance, you may want to have your step retrieve the new changes since the last execution and it is sufficient to store the last retrieved items and only return the new items since.

Internal states are just normal resources (of resource type state) and they have for path `<executer_of_flow>/<flow_path>/<script_path>` when ensure their uniqueness relative to the executer, the flow and the script path.